crates/bevy_ui/src/widget/text.rs

```diff
@@ -78,19 +104,34 @@ pub fn text_system(
     let mut new_queue = Vec::new();
     let mut query = text_queries.p2();
     for entity in queued_text.entities.drain(..) {
-        if let Ok((text, style, mut calculated_size)) = query.get_mut(entity) {
+        if let Ok((text, style, mut calculated_size, parent)) = query.get_mut(entity) {
+            let mut scale_factor_width = scale_factor;
+            let mut scale_factor_height = scale_factor;
+            if let Some(parent) = parent {
+                if let Ok(style) = style_query.get(parent.get()) {
+                    if let Val::Percent(percentage) = style.size.width {
+                        scale_factor_width = percentage as f64 / (scale_factor * 100.);
+                    }
+                    if let Val::Percent(percentage) = style.size.height {
+                        scale_factor_height = percentage as f64 / (scale_factor * 100.);
+                    }
+                }
+            }
```

Comment:
It feels very strange to me that you're changing the scale factor instead of the size constraint
That could explain why it doesn't render as you expect on screen with different scale factor than 1

Reply:
yeah I think that was the wrong approach by me, I'll update it to target the style.size instead